routes/associations.go

```diff
@@ -0,0 +1,1036 @@
+		_AddInternalServerError(ww, "CreateUserAssociation: problem getting public key for the target user")
+		return
+	}
+
+	// Parse AppPublicKeyBytes from AppPublicKeyBase58Check
+	if requestData.AppPublicKeyBase58Check == "" {
+		_AddBadRequestError(ww, "CreateUserAssociation: must provide an AppPublicKeyBase58Check")
+		return
+	}
+	appPublicKeyBytes, _, err := fes.GetPubKeyAndProfileEntryForUsernameOrPublicKeyBase58Check(
+		requestData.AppPublicKeyBase58Check,
+		utxoView,
+	)
+	if err != nil {
+		_AddInternalServerError(ww, "CreateUserAssociation: problem getting public key for the app")
+		return
+	}
+
+	// Validate AssociationType.
+	if requestData.AssociationType == "" {
+		_AddBadRequestError(ww, "CreateUserAssociation: must provide an AssociationType")
+		return
+	}
+
+	// Validate AssociationValue.
+	if requestData.AssociationValue == "" {
+		_AddBadRequestError(ww, "CreateUserAssociation: must provide an AssociationValue")
+		return
+	}
+
+	// Parse ExtraData.
+	extraData, err := EncodeExtraDataMap(requestData.ExtraData)
+	if err != nil {
+		_AddBadRequestError(ww, "CreateUserAssociation: invalid ExtraData provided")
+		return
+	}
+
+	// Compute the additional transaction fees as specified
+	// by the request body and the node-level fees.
+	additionalOutputs, err := fes.getTransactionFee(
+		lib.TxnTypeCreateUserAssociation,
+		transactorPublicKeyBytes,
+		requestData.TransactionFees,
+	)
+	if err != nil {
+		_AddBadRequestError(ww, "CreateUserAssociation: specified TransactionFees are invalid")
+		return
+	}
+
+	// Create transaction.
+	txn, totalInput, changeAmount, fees, err := fes.blockchain.CreateCreateUserAssociationTxn(
+		transactorPublicKeyBytes,
+		&lib.CreateUserAssociationMetadata{
+			TargetUserPublicKey: lib.NewPublicKey(targetUserPublicKeyBytes),
+			AppPublicKey:        lib.NewPublicKey(appPublicKeyBytes),
+			AssociationType:     []byte(requestData.AssociationType),
+			AssociationValue:    []byte(requestData.AssociationValue),
+		},
+		extraData,
+		requestData.MinFeeRateNanosPerKB,
+		fes.backendServer.GetMempool(),
+		additionalOutputs,
+	)
+	if err != nil {
+		_AddInternalServerError(ww, fmt.Sprintf("CreateUserAssociation: problem creating txn: %v", err))
+		return
+	}
+
+	// Construct response.
+	txnBytes, err := txn.ToBytes(true)
+	if err != nil {
+		_AddInternalServerError(ww, "CreateUserAssociation: problem encoding txn to bytes")
+		return
+	}
+	res := AssociationTxnResponse{
+		SpendAmountNanos:  totalInput - changeAmount - fees,
+		TotalInputNanos:   totalInput,
+		ChangeAmountNanos: changeAmount,
+		FeeNanos:          fees,
+		Transaction:       txn,
+		TransactionHex:    hex.EncodeToString(txnBytes),
+		TxnHashHex:        txn.Hash().String(),
+	}
+	if err = json.NewEncoder(ww).Encode(res); err != nil {
+		_AddInternalServerError(ww, "CreateUserAssociation: problem encoding response as JSON")
+		return
+	}
+}
+
+func (fes *APIServer) DeleteUserAssociation(ww http.ResponseWriter, req *http.Request) {
```

Comment:
do we also what to support a DeleteByAttributes?

Reply:
IMO no. GetByID mirrors DeleteByID. GetByAttributes is to support querying. If required, a user could do a GetByAttributes query then a subsequent DeleteByID. No sense in adding that complexity to core or backend IMO.